File: examples/android/MSDKSampleAppJava/lib/src/main/java/msdk/java/messages/OutOfBandInvitation.java
package msdk.java.messages;

import androidx.annotation.NonNull;

import org.json.JSONObject;

import msdk.java.handlers.Connections;

public class OutOfBandInvitation {
    public String invitation;
    public JSONObject attachment;
    public Connections.ConnectionMetadata userMeta;
    public String existingConnection;

    public OutOfBandInvitation(
            String invitation,
            JSONObject attach,
            Connections.ConnectionMetadata userMeta,
            String existingConnection
    ) {
        this.invitation = invitation;
        this.attachment = attach;
        this.existingConnection = existingConnection;
        this.userMeta = userMeta;
    }

    public static class OutOfBandInviteBuilder {
        private String invitation;
        private JSONObject attachment;
        private String existingConnection;
        private Connections.ConnectionMetadata userMeta;

        private OutOfBandInviteBuilder() {
        }

        public @NonNull
        OutOfBandInviteBuilder withInvitation(@NonNull String invitation) {
            this.invitation = invitation;
            return this;
        }

        public @NonNull
        OutOfBandInviteBuilder withAttachment(@NonNull JSONObject attach) {
            this.attachment = attach;
            return this;
        }

        public @NonNull
        OutOfBandInviteBuilder withUserMeta(Connections.ConnectionMetadata userMeta) {
            this.userMeta = userMeta;
            return this;
        }

        public @NonNull
        OutOfBandInviteBuilder withExistingConnection(String existingConnection) {
            this.existingConnection = existingConnection;
            return this;
        }

        public @NonNull
        OutOfBandInvitation build() {
            return new OutOfBandInvitation(
                    invitation,
                    attachment,
                    userMeta,
                    existingConnection
            );
        }
    }

    public static OutOfBandInviteBuilder builder() {
        return new OutOfBandInviteBuilder();
    }
}

File: examples/android/MSDKSampleAppKotlin/app/src/main/java/msdk/kotlin/sample/messages/OutOfBandInvitation.kt
package msdk.kotlin.sample.messages

import msdk.kotlin.sample.handlers.Connections.ConnectionMetadata
import org.json.JSONObject

class OutOfBandInvitation(
    var invitation: String?,
    var attachment: JSONObject?,
    var userMeta: ConnectionMetadata?,
    var existingConnection: String?
) {

    class OutOfBandInviteBuilder {
        private var invitation: String? = null
        private var attachment: JSONObject? = null
        private var existingConnection: String? = null
        private var userMeta: ConnectionMetadata? = null
        fun withInvitation(invitation: String): OutOfBandInviteBuilder {
            this.invitation = invitation
            return this
        }

        fun withAttachment(attach: JSONObject): OutOfBandInviteBuilder {
            attachment = attach
            return this
        }

        fun withUserMeta(userMeta: ConnectionMetadata?): OutOfBandInviteBuilder {
            this.userMeta = userMeta
            return this
        }

        fun withExistingConnection(existingConnection: String?): OutOfBandInviteBuilder {
            this.existingConnection = existingConnection
            return this
        }

        fun build(): OutOfBandInvitation {
            return OutOfBandInvitation(
                invitation,
                attachment,
                userMeta,
                existingConnection
            )
        }
    }

    companion object {
        fun builder(): OutOfBandInviteBuilder {
            return OutOfBandInviteBuilder()
        }
    }

}